src/recode.jl

```diff
@@ -52,23 +53,27 @@ A user defined type could override this method to define an appropriate test fun
 optimize_pair(pair::Pair) = pair
 optimize_pair(pair::Pair{<:AbstractArray}) = Set(pair.first) => pair.second
 
-function recode!(dest::AbstractArray{T}, src::AbstractArray, default::Any, pairs::Pair...) where {T}
+function recode!(dest::AbstractArray, src::AbstractArray, default::Any, pairs::Pair...)
     if length(dest) != length(src)
         throw(DimensionMismatch("dest and src must be of the same length (got $(length(dest)) and $(length(src)))"))
     end
 
-    opt_pairs = map(optimize_pair, pairs)
+    opt_pairs = optimize_pair.(pairs)
 
+    _recode!(dest, src, default, opt_pairs)
+end
+
+function _recode!(dest::AbstractArray{T}, src::AbstractArray, default, pairs) where {T}
```

Comment:
For clarity (another solution would be to use varargs/splatting again):
```suggestion
function _recode!(dest::AbstractArray{T}, src::AbstractArray, default,
                  pairs::NTuple{<:Any, Pair}) where {T}
```

Reply:
I used a tuple instead of a splat here because the former gives better type stability, but the NTuple is a good catch